packages/flutter_tools/lib/src/commands/assemble.dart

```diff
@@ -263,17 +263,25 @@ class AssembleCommand extends FlutterCommand {
       dartDefines = argumentResults[FlutterOptions.kDartDefinesOption] as List<String>;
     }
     if (argumentResults.wasParsed(FlutterOptions.kDartDefineFromFileOption)) {
-      final String? configJsonPath = stringArg(FlutterOptions.kDartDefineFromFileOption);
-      if (configJsonPath != null && globals.fs.isFileSync(configJsonPath)) {
-        final String configJsonRaw = globals.fs.file(configJsonPath).readAsStringSync();
-        try {
-          (json.decode(configJsonRaw) as Map<String, dynamic>).forEach((String key, dynamic value) {
-            dartDefines.add('$key=$value');
-          });
-        } on FormatException catch (err) {
-          throwToolExit('Json config define file "--${FlutterOptions.kDartDefineFromFileOption}=$configJsonPath" format err, '
-              'please fix first! format err:\n$err');
+      final List<String> configJsonPaths = stringsArg(FlutterOptions.kDartDefineFromFileOption);
```

Comment:
This is also tech debt from a previous change, but extending this code highlights the fact that this code is not dry, and changes to this block of code also need to be made to flutter_command.dart. Can you instead move this code to a helper function in `flutter_command.dart` (since `AssembleCommand extends FlutterCommand` it should be accessible in this file) that parses both the `--dart-define-from-file` and `--dart-define` options and returns a single `List<String>` of dartDefines?